anta/tests/snmp.py

```diff
@@ -350,3 +352,103 @@ def test(self) -> None:
             self.result.is_success()
         else:
             self.result.is_failure(f"The following SNMP error counters are not found or have non-zero error counters:\n{error_counters_not_ok}")
+
+
+class VerifySnmpUser(AntaTest):
+    """Verifies the SNMP user configurations for specified version(s).
+
+    - Verifies that the valid user name and group name.
+    - Ensures that the SNMP v3 security model, the user authentication and privacy settings aligning with version-specific requirements.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the provided SNMP user and all specified parameters are correctly configured.
+    * Failure: The test will fail if the provided SNMP user is not configured or specified parameters are not correctly configured.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.snmp:
+      - VerifySnmpUser:
+          users:
+            - username: test
+              group_name: test_group
+              security_model: v3
+              authentication_type: MD5
+              priv_type: AES-128
+    ```
+    """
+
+    name = "VerifySnmpUser"
+    description = "Verifies the SNMP user configurations for specified version(s)."
+    categories: ClassVar[list[str]] = ["snmp"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show snmp user", revision=1)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifySnmpUser test."""
+
+        users: list[SnmpUser]
+        """List of SNMP users."""
+
+        class SnmpUser(BaseModel):
+            """Model for a SNMP User."""
+
+            username: str
+            """SNMP user name."""
+            group_name: str
+            """SNMP group for the user."""
+            security_model: SnmpVersion
+            """SNMP protocol version.."""
+            authentication_type: HashingAlgorithms | None = None
+            """User authentication settings."""
+            priv_type: EncryptionAlgorithms | None = None
+            """User privacy settings."""
+
+            @model_validator(mode="after")
+            def validate_inputs(self: BaseModel) -> BaseModel:
+                """Validate the inputs provided to the SnmpUser class."""
+                if self.security_model in ["v1", "v2c"] and (self.authentication_type or self.priv_type) is not None:
+                    msg = "SNMP versions 1 and 2c, do not support encryption or advanced authentication."
```

Comment:
So we should use the validator in case if a input is required for a case. So i think if a user provide the the version as v3 in that case we should raise the error that encryption and authentication is required.